alchemy/src/neon/database.ts

```diff
@@ -0,0 +1,285 @@
+import type { Context } from "../context.ts";
+import { Resource } from "../resource.ts";
+import { handleApiError } from "./api-error.ts";
+import { createNeonApi, type NeonApiOptions, type NeonApi } from "./api.ts";
+import type { NeonProject } from "./project.ts";
+import type { NeonBranch } from "./branch.ts";
+
+/**
+ * Properties for creating or updating a Neon database
+ */
+export interface NeonDatabaseProps extends NeonApiOptions {
+  /**
+   * The project containing the branch
+   */
+  project: string | NeonProject;
+
+  /**
+   * The branch where the database will be created
+   */
+  branch: string | NeonBranch;
+
+  /**
+   * Name of the database
+   */
+  name: string;
+
+  /**
+   * Name of the role that will own the database
+   */
+  ownerName: string;
+
+  /**
+   * Whether to adopt an existing database if it already exists
+   * @default false
+   */
+  adopt?: boolean;
+}
+
+/**
+ * A Neon database within a branch
+ */
+export interface NeonDatabase
+  extends Resource<"neon::Database">,
+    Omit<NeonDatabaseProps, "apiKey"> {
+  /**
+   * The database ID
+   */
+  id: number;
+
+  /**
+   * The ID of the project containing this database
+   */
+  projectId: string;
+
+  /**
+   * The ID of the branch containing this database
+   */
+  branchId: string;
+
+  /**
+   * Name of the database
+   */
+  name: string;
+
+  /**
+   * Name of the role that owns the database
+   */
+  ownerName: string;
+
+  /**
+   * Time at which the database was created
+   */
+  createdAt: string;
+
+  /**
+   * Time at which the database was last updated
+   */
+  updatedAt: string;
+}
+
+/**
+ * Creates a Neon database within a branch.
+ *
+ * @example
+ * // Create a basic database with default owner:
+ * const database = await NeonDatabase("my-database", {
+ *   project: project,
+ *   branch: branch,
+ *   name: "myapp_db",
+ *   ownerName: "neondb_owner"
+ * });
+ *
+ * @example
+ * // Create a database with a custom owner role:
+ * const database = await NeonDatabase("custom-db", {
+ *   project: project,
+ *   branch: branch,
+ *   name: "analytics_db",
+ *   ownerName: "analytics_user"
+ * });
+ *
+ * @example
+ * // Adopt an existing database if it already exists:
+ * const database = await NeonDatabase("existing-db", {
+ *   project: project,
+ *   branch: branch,
+ *   name: "legacy_db",
+ *   ownerName: "neondb_owner",
+ *   adopt: true
+ * });
+ */
+export const NeonDatabase = Resource(
```

Comment:
I noticed project.ts has a NeonDatabase interface, can we make sure all NeonDatabase code and types are here in this file and imported into project.ts